1. Provided by Lars-Peter Clausen <lars@metafoo.de>
   http://dev.openwrt.org/browser/packages/Xorg/driver/xf86-input-tslib/patches
2. Refreshed for other debian-specific patches

--- xf86-input-tslib-0.0.5.orig/src/tslib.c
+++ xf86-input-tslib-0.0.5/src/tslib.c
@@ -48,6 +48,7 @@
 #include <exevents.h>		/* Needed for InitValuator/Proximity stuff */
 #include <X11/keysym.h>
 #include <mipointer.h>
+#include <randrstr.h>
 
 #include <tslib.h>
 #include <sys/time.h>
@@ -138,6 +139,8 @@
 	struct ts_sample samp;
 	int ret;
 	int x,y;
+	ScrnInfoPtr pScrn = xf86Screens[priv->screen_num];
+	Rotation rotation = RRGetRotation(pScrn->pScreen);
 	struct timeval now;
 
 	while((ret = ts_read(priv->ts, &samp, 1)) == 1) {
@@ -160,6 +163,23 @@
 			default:		break;
 			}
 
+			tmp_x = samp.x;
+
+			switch(rotation) {
+			case RR_Rotate_90:
+				samp.x = (priv->height - samp.y - 1) * priv->width / priv->height;
+				samp.y = tmp_x * priv->height / priv->width;
+				break;
+			case RR_Rotate_180:
+				samp.x = priv->width - samp.x - 1;
+				samp.y = priv->height - samp.y - 1;
+				break;
+			case RR_Rotate_270:
+				samp.x = samp.y * priv->width / priv->height;
+				samp.y = (priv->width - tmp_x - 1) * priv->height / priv->width;
+				break;
+			}
+
 			priv->lastx = samp.x;
 			priv->lasty = samp.y;
 			x = samp.x;
